main/src/ca/uwaterloo/flix/language/ast/Ast.scala

```diff
@@ -360,7 +362,12 @@ object Ast {
   /**
     * Contains the edges held in a single constraint.
     */
-  case class MultiEdge(head: Name.Pred, positives: Set[(Name.Pred, SourceLocation)], negatives: Set[(Name.Pred, SourceLocation)])
+  case class MultiEdge(head: (Name.Pred, Type), positives: Vector[MultiElement], negatives: Vector[MultiElement])
+
+  /**
+    * Represents a body predicate on a MultiEdge.
+    */
+  type MultiElement = (Name.Pred, Type, SourceLocation)
```

Comment:
Turn into a case class, and can we find a better name? It is not a Multi-anything given that there is only one, right?